azurerm/resource_arm_scheduler_job_collection.go

```diff
@@ -258,20 +271,29 @@ func expandAzureArmSchedulerJobCollectionQuota(d *schema.ResourceData) *schedule
 	return nil
 }
 
-func flattenAzureArmSchedulerJobCollectionQuota(quota *scheduler.JobCollectionQuota) []interface{} {
+func flattenAzureArmSchedulerJobCollectionQuota(quota *scheduler.JobCollectionQuota, oldBlock []interface{}) []interface{} {
 
 	if quota == nil {
 		return nil
 	}
 
+	//todo remove once max_retry_interval is retired
+	maxRecurrenceField := "max_recurrence_interval"
+	if len(oldBlock) > 0 {
+		b := oldBlock[0].(map[string]interface{})
+
+		if v, ok := b["max_retry_interval"].(int); ok && v > 0 {
+			maxRecurrenceField = "max_retry_interval"
+		} //else max_recurrence_interval is in use, or we default to max_recurrence_interval
+	}
```

Comment:
rather than doing this, we can instead just assign both of these values, since one's deprecated